firedrake/preconditioners/asm.py

```diff
@@ -61,14 +61,15 @@ def initialize(self, pc):
             opts = PETSc.Options(asmpc.getOptionsPrefix())
```

Comment:
I'm a bit concerned/confused by the double-definition of `opts` here, one with `self.prefix` and one with `asmpc.getOptionsPrefix()`.  Since the latter is `self.prefix + "sub_"`, I don't think the two lines below defining `ordering` are equivalent.  Are they?

Reply:
You are right at pointing this naming conflict. But I think `asmpc.setOptionsPrefix(self.prefix + "sub_")` was a mistake. This is why we need to pass the infamous `star_sub_sub_` prefix, and I think we could be fine if we just set `asmpc.setOptionsPrefix(self.prefix)` as we have in patch.py.

Reply:
This looks like cleaner code now - thanks!  Do we want to resolve the `star_sub_sub_` here or in another PR or not at all?

Reply:
I'll open another PR, this would definitely break every code that passes options to set the patch solvers!